vault/request_handling.go

```diff
@@ -675,6 +675,10 @@ func (c *Core) handleCancelableRequest(ctx context.Context, req *logical.Request
 		resp, auth, err = c.handleRequest(ctx, req)
 	}
 
+	if err != nil && c.requestResponseCallback != nil {
```

Comment:
don't we want to call this only on non login requests i.e. up 3 lines?

Reply:
I don't know?  To me it seems like login requests are just requests, so why wouldn't we want a response callback to be invoked for them?

Reply:
Oh gotcha maybe I misunderstood the comment that was added for `requestResponseCallback` which said we wanted to call it on every non-login request.

Reply:
We currently have `approle` response schemas defined but for other login requests, the test will likely fail. If that's the case, perhaps add a `// TODO` note to move this up after all the response schemas are defined.

Reply:
@averche it should be simple enough to test if it fails, can you point me to another login request method?